bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Tags added for node '{}'.", node.getPath());
+            }
+        }
+    }
+
+    /**
+     * Set generic array property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setArrayProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray rawValues = (JSONArray) json.get(key);
+        Object firstVal = rawValues.get(0);
+        Value[] propertyValues = new Value[rawValues.length()];
+
+        if (firstVal instanceof Boolean) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getBoolean(i));
+            }
+        } else if (firstVal instanceof Double) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getDouble(i));
+            }
+        } else if (firstVal instanceof Integer) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getInt(i));
+            }
+        } else if (firstVal instanceof Long) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getLong(i));
+            }
+        } else {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getString(i));
+            }
+        }
+
+        node.setProperty(key, propertyValues);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Array property '{}' added for node '{}'.", key, node.getPath());
+        }
+    }
+
+    /**
+     * Set JCR Data property to a temporary binary for the property's node.
+     * Also, property 'jcr:data' is a mandatory field.
+     * @param node Node
+     * @param rawResponseLastModified String
+     * @throws IOException exception
+     * @throws RepositoryException exception
+     */
+    private void setJcrDataProperty(final Node node, final String rawResponseLastModified) throws IOException, RepositoryException {
+        if (node.hasProperty(JcrConstants.JCR_LASTMODIFIED) && node.hasProperty(JcrConstants.JCR_DATA)
+                && StringUtils.isNotEmpty(rawResponseLastModified)) {
+
+            String nodeLastModified = node.getProperty(JcrConstants.JCR_LASTMODIFIED).getString();
+            Calendar responseLastModified = getFormattedDate(rawResponseLastModified);
+
+            if (responseLastModified != null && nodeLastModified.equals(this.valueFactory.createValue(responseLastModified).getString())) {
+                return;
```

Comment:
This is probably worth a `debug` log so someone could determine why a particular asset isnt getting temporary binaries.